src/principal/service.ts

```diff
@@ -46,20 +46,35 @@ export class PrincipalService {
 
   }
 
-  async findAll(type: 'user'): Promise<User[]>;
+  async findAll(type: 'user', page: number): Promise<User[]>;
   async findAll(type: 'group'): Promise<Group[]>;
   async findAll(type: 'app'): Promise<App[]>;
   async findAll(): Promise<Principal[]>;
-  async findAll(type?: PrincipalType): Promise<Principal[]> {
+  async findAll(type?: PrincipalType, page?: number): Promise<Principal[]> {
 
     this.privileges.require('a12n:principals:list');
     const filters: Record<string, any> = {};
     if (type) {
       filters.type = userTypeToInt(type);
     }
 
-    const result = await db('principals')
-      .where(filters);
+    let result: PrincipalsRecord[] = [];
+
+    if(type && page !== undefined){
```

Comment:
Could also be written as: 

```suggestion
    if (type && page){
```

Because `undefined` is falsy

Reply:
oh, but `0` is also falsy, but that's valid input